Dart/src/com/jetbrains/lang/dart/ide/completion/DartServerCompletionContributor.java

```diff
@@ -93,33 +113,34 @@ private static LookupElement createLookupElement(Project project, CompletionSugg
         icon = applyOverlay(icon, element.isConst(), AllIcons.Nodes.FinalMark);
         lookup = lookup.withIcon(icon);
       }
-      // insert
-      // TODO(scheglov) insert () with arguments
-      //lookup = lookup.withInsertHandler(new InsertHandler<LookupElement>() {
-      //  @Override
-      //  public void handleInsert(InsertionContext context, LookupElement item) {
-      //    context.getDocument().insertString(context.getStartOffset(), "()");
-      //  }
-      //});
+      // insert handler
+      lookup = lookup.withInsertHandler(new InsertHandler<LookupElement>() {
+        @Override
+        public void handleInsert(InsertionContext context, LookupElement item) {
```

Comment:
As a further improvement take a look at AutoPopupController#autoPopupParameterInfo. It shows parameter info tooltip - very handy when we are filling parameters after selecting method from completion list.